nwb_conversion_tools/json_schema_utils.py

```diff
@@ -64,22 +67,29 @@ def get_schema_from_method_signature(class_method, exclude=None):
         list="array"
     )
 
-    for param in inspect.signature(class_method).parameters.values():
-        if param.name not in exclude + ['self']:
+    for param_name, param in inspect.signature(class_method).parameters.items():
+        if param_name not in exclude:
             if param.annotation:
-                param_type = annotation_json_type_map[str(param.annotation).split("'")[1]]
+                param_type = [
+                    annotation_json_type_map[anno] for arg_types in param.annotation.__args__
+                    for anno in [str(arg_types).split("'")[1]] if anno in annotation_json_type_map
```

Comment:
Purposefully not using `.get()` here to avoid adding excess items to this list that would have to be removed or ignored afterward.

Reply:
I don't understand this comprehension. Can you break it out for me? Why are we casting arg_types as a string?

Reply:
It's essentially the same previous 3 lines of pre-Union version code. It always used to convert it to a string to extract the class name, but I just realized I could simply do `arg_types.__name__`. Fixed that and cleaned it up a tad on the latest commit.

But otherwise, the `Union` usage allows `.__args__` to be an iterable, so that's why that for loop is in the compression. For each argument in that iterable it passes it through the json type map, but only if that is a valid key (bypassing the need for the previous intersection stuff).